Bottom: f8d276153f198899dee765bd7cb1b588021de09b
Top:    5b86dca81bff773b77658c3aed234f6dd10b8410
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-11 04:52:20 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index dbb3d12..c897667 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -236,6 +236,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
